src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs

```diff
@@ -1238,18 +1238,18 @@ public sealed override async Task<HttpResponseMessage> SendAsync(HttpRequestMess
             {
                 http2Stream?.Dispose();
 
-                if (e is IOException ioe)
+                if (e is IOException)
                 {
-                    throw new HttpRequestException(SR.net_http_client_execution_error, ioe);
+                    throw new HttpRequestException(SR.net_http_client_execution_error, e);
                 }
                 else if (e is ObjectDisposedException)
                 {
-                    throw new HttpRequestException(SR.net_http_client_execution_error);
+                    throw new HttpRequestException(SR.net_http_client_execution_error, e);
                 }
                 else if (e is Http2ProtocolException)
                 {
                     // ISSUE 31315: Determine if/how to expose HTTP2 error codes
-                    throw new HttpRequestException(SR.net_http_client_execution_error);
+                    throw new HttpRequestException(SR.net_http_client_execution_error, e);
```

Comment:
Http2ProtocolException is not a public type. See referenced issue. I don't think it does any good to surface the exception if it's not a public type, does it?

Reply:
It is useful because it will show that exception and the exception message and stack trace in all diagnostics.

Yes, it is not a public type. But it doesn't matter.  For example, HttpClient using WinHttpHandler will show the WinHttpException as the inner exception for top-level HttpRequestException.  WinHttpException is not a public type either.  But it is still very useful to see it in diagnostics, stack traces, etc.

Reply:
Ok